src/components/specific/versioning/versioning-main/VersioningMain.vue

```diff
@@ -0,0 +1,188 @@
+<template>
+  <div class="versioning-main">
+    <transition name="fade">
+      <template v-if="isSafeZone">
+        <div class="versioning-main__safe-zone">
+          <VersioningSafeZone @close="onSafeZone" />
+        </div>
+      </template>
+    </transition>
+    <template v-if="loading">
+      <BIMDataLoading
+        :message="$t('Versioning.loading.message')"
+        :subMessage="$t('Versioning.loading.subMessage')"
+      />
+    </template>
+    <div class="versioning-main__content" :class="{ safeZone: isSafeZone }">
+      <div class="versioning-main__content__header">
+        <div class="versioning-main__content__header__left-side">
+          <BIMDataIcon
+            name="versioning"
+            fill
+            color="primary"
+            size="s"
+            margin="2.5px 0 0 0"
+          />
+          <span>{{ $t("Versioning.title") }}</span>
+        </div>
+        <div class="versioning-main__content__header__right-side">
+          <BIMDataButton ghost rounded icon @click="$emit('close')">
+            <BIMDataIcon name="close" size="xxs" fill color="granite-light" />
+          </BIMDataButton>
+        </div>
+      </div>
+
+      <div class="versioning-main__content__add-version">
+        <FileUploadButton
+          :disabled="
+            (!project.isAdmin && currentFolder.userPermission < 100) ||
+            spaceSubInfo.remainingTotalSize <= 0
+          "
+          width="100%"
+          icon="close"
+          iconSize="xxxs"
+          :iconRotate="45"
+          :textButton="$t('Versioning.AddVersion')"
+          @upload="addVersion"
+        />
+      </div>
+      <template v-if="hasPrevVersions === false">
+        <div class="versioning-main__content__empty-area">
+          <div class="versioning-main__content__empty-area__info">
+            <img src="/static/versioning/versioning-empty-history.svg" />
+            <span>{{ $t("Versioning.emptyArea") }}</span>
+          </div>
+        </div>
+      </template>
+      <template v-else>
+        <div class="versioning-main__content__list">
+          <template v-for="(doc, index) of allDocVersions" :key="index">
```

Comment:
I see an **index** as `key` here.
Did you check that it does not result in the same bug that we discuss last time ?
(As a reminder: the UI was not properly updated when deleting the first element due to the way Vue uses `key` attribute internally to identify nodes).